components/builder-http-gateway/src/http/helpers.rs

```diff
@@ -429,6 +431,52 @@ pub fn get_optional_session_id(req: &mut Request) -> Option<u64> {
     }
 }
 
+pub fn generate_origin_encryption_keys(
+    req: &mut Request,
+    session: Session,
+    origin: Origin,
+) -> NetResult<()> {
+    let mut public_request = OriginPublicEncryptionKeyCreate::new();
+    let mut private_request = OriginPrivateEncryptionKeyCreate::new();
+    let mut public_key = OriginPublicEncryptionKey::new();
+    let mut private_key = OriginPrivateEncryptionKey::new();
+    public_key.set_owner_id(session.get_id());
+    private_key.set_owner_id(session.get_id());
+    public_key.set_name(origin.get_name().to_string());
+    public_key.set_origin_id(origin.get_id());
+    private_key.set_name(origin.get_name().to_string());
+    private_key.set_origin_id(origin.get_id());
+
+    let pair = BoxKeyPair::generate_pair_for_origin(origin.get_name())
+        .expect("failed to generate origin encryption key pair");
```

Comment:
Handle failure instead of expect?

Reply:
In fact, looking through the code that's already in this crate, there are `expect` and `unwrap` calls everywhere, so this at least seems consistent to me.  https://github.com/habitat-sh/builder/issues/219 has been created to track this.